packages/front-end/pages/saved-groups.tsx

```diff
@@ -48,6 +52,47 @@ export default function SavedGroupsPage() {
 
   const [auditModal, setAuditModal] = useState(false);
 
+  const { refreshOrganization } = useUser();
+
+  const permissions = usePermissions();
+  const { apiCall } = useAuth();
+  const attributeSchema = useAttributeSchema();
+
+  useEffect(() => {
+    // Not using $groups attribute in a any saved groups
+    if (
+      !savedGroups?.some(
+        (g) => g.type === "condition" && g.condition?.includes("$groups")
+      )
+    ) {
+      return;
+    }
+
+    // Already has $groups attribute
+    if (attributeSchema.some((a) => a.property === "$groups")) return;
+
+    // If user has permissions to manage attributes, auto-add $groups attribute
```

Comment:
I noticed the 'auto-add' endpoint added to the organization router/controller but having trouble understanding it. It seems like a legacy way of storing saved groups. Is that correct? Could you explain a little bit about it?

Reply:
We had the concept of "runtime" saved groups.  It would look something like this:

```js
{
  id: "sg_abc123",
  organization: "org_abc123",
  source: "runtime",
  attributeKey: "admins"
}
```

When we rendered this into the SDK payload, we would convert it to a condition looking for this special `$groups` attribute:

```js
{
  defaultValue: false,
  rules: [
    {
      condition: { "$groups": {"$elemMatch": {"$eq": "admins"}}},
      force: true
    }
  ]
}
```

And we told people to pass that `$groups` attribute into their SDK:

```ts
new GrowthBook({
  attributes: {
    $groups: ["admins", "beta-users"]
  }
})
```

Since this was a special attribute, we didn't require people to add it to the **SDK Configuration -> Attributes** page to be able to use it.

Now that we switched to use the `<ConditionInput>` component for these groups, it's a pretty bad experience if you go try and edit one of these old runtime groups.  If that attribute doesn't exit, it defaults to the advanced JSON editing view which is hard to use.

So this is a little utility endpoint to auto-add a `$groups` attribute to their org if it's missing and they had previously set up a runtime group.  If it fails to run for whatever reason, it's not the end of the world.  The user experience will just be slightly degraded until they manually define the attribute in the UI.